### PR TITLE
Ineffassign fixes

### PR DESCRIPTION
**Pull request by alena1108**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

Some places had ineffectual assignments. In a couple of places returned error check was missing

**Which issue(s) this PR fixes**:

Fixes #

N/A

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/volume/util/fsquota/quota_linux.go
Comment (mattjmcnaughton) [inline pkg/volume/util/fsquota/quota_linux.go]: Should we check `if err != nil` and then check if `projid != dirQuotaMap...`?

File: pkg/volume/util/fsquota/quota_linux.go
Comment (mattjmcnaughton) [inline pkg/volume/util/fsquota/quota_linux.go]: Should we be checking `if err != nil` before we attempt to check `supportsQuota`?

File: pkg/volume/azure_file/azure_file.go
Comment (mattjmcnaughton) [inline pkg/volume/azure_file/azure_file.go]: Just to confirm, do we intentionally want to change the behavior here? In other words, are we comfortable that an `err` in `getSecretNameAndNamespace` will now cause `newMounterInternal` to return an error?

It could be worth proposing this change in a separate pr :)

File: pkg/volume/azure_dd/azure_mounter.go
Comment (mattjmcnaughton) [inline pkg/volume/azure_dd/azure_mounter.go]: Can you share more about how we can be confident this assignment is safe to delete?

File: pkg/proxy/winkernel/hnsV1.go
Comment (mattjmcnaughton) [inline pkg/proxy/winkernel/hnsV1.go]: I think that for v1 (and the v2 code below) we may want use the error included in the return statement? I also think we can skip the `klog.Errorf`... typically its the responsibility of whatever calls the func to handle the error.

**Review by mattjmcnaughton — COMMENTED** [review]
Thanks for your work on this clean up :)

**Comment by mattjmcnaughton** [comment]
Also, welcome to k8s - congrats on your first pr :)

/ok-to-test

File: pkg/volume/util/fsquota/quota_linux.go
Comment (alena1108) [reply pkg/volume/util/fsquota/quota_linux.go]: @mattjmcnaughton good point. Was just assuming that in case of err != nil `projid != dirQuotaMap[path]` condition will be true, and didn't want to alter the existing behavior. I'll do error check beforehand.

File: pkg/volume/azure_file/azure_file.go
Comment (alena1108) [reply pkg/volume/azure_file/azure_file.go]: @mattjmcnaughton seeing azure e2e tests failing, I think I'll go with the safer fix for now, and will just log the error here, and in other places, w/o them being returned. Can you suggest what level of log you think is appropriate here

File: pkg/volume/util/fsquota/quota_linux.go
Comment (alena1108) [reply pkg/volume/util/fsquota/quota_linux.go]: yup, will address is similar to https://github.com/kubernetes/kubernetes/pull/87462#discussion_r370753862

File: pkg/volume/azure_dd/azure_mounter.go
Comment (alena1108) [reply pkg/volume/azure_dd/azure_mounter.go]: @mattjmcnaughton sure. mountPoint's value previously assigned on line 105 is not being read anywhere after this point. It gets reassigned anew on this line https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/azure_dd/azure_mounter.go#L139

File: pkg/proxy/winkernel/hnsV1.go
Comment (alena1108) [reply pkg/proxy/winkernel/hnsV1.go]: Was following the existing pattern on line 73 of the same file; but will fix!

File: pkg/volume/azure_dd/azure_mounter.go
Comment (mattjmcnaughton) [reply pkg/volume/azure_dd/azure_mounter.go]: Hmm - I'm far from an expert on this part of the code base... but is `mountPoint = true` an truthful reflection of the state of the world at line 105 and after? If yes, it still feels useful to have be accurate, even if its not being used now.

File: pkg/volume/azure_file/azure_file.go
Comment (mattjmcnaughton) [reply pkg/volume/azure_file/azure_file.go]: I defer to the Azure folks on what's best :) I don't have a ton of context. But I definitely agree that's less risky :)

File: pkg/volume/azure_dd/azure_mounter.go
Comment (alena1108) [reply pkg/volume/azure_dd/azure_mounter.go]: @mattjmcnaughton I can put it back if it's a deal breaker

File: pkg/volume/azure_file/azure_file.go
Comment (alena1108) [reply pkg/volume/azure_file/azure_file.go]: @mattjmcnaughton having error logged as a part of this pr wouldn't hurt, and could useful for debugging at least :) Once it's merged, I can open a separate pr with the error returned, as you've suggested earlier + will tag Azure folks there.

**Comment by alena1108** [comment]
@mattjmcnaughton addressed all your comments + `pull-kubernetes-e2e-gce` had finally passed :crossed_fingers:

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

/assign @lavalamp

for final approval!

**Comment by lavalamp** [comment]
Honestly I'm nervous about this many potentially subtle error handling changes in one PR. Sorry for asking for work, but how do you feel about splitting this into a PR per package fixed?

Also I recommend adding a comment stating it's for unspecified backwards compatibility concerns when the response to an error is log-and-continue, so that future readers know why.

**Comment by alena1108** [comment]
@lavalamp 

> Sorry for asking for work, but how do you feel about splitting this into a PR per package fixed?

Np at all. Let me know if ok with me splitting it as follows, and I'll close this pr in favor of new ones (and will comment to the log-and-continue places):
```
pkg/volume
pkg/util
pkg/kubelet + pkg/controller // both one file/one liner non-error handling related changes
pkg/proxy
```

**Comment by lavalamp** [comment]
SGTM, thanks!

**Comment by alena1108** [comment]
Closing in favor of following per-package PRs:


https://github.com/kubernetes/kubernetes/pull/87708
https://github.com/kubernetes/kubernetes/pull/87710
https://github.com/kubernetes/kubernetes/pull/87711
https://github.com/kubernetes/kubernetes/pull/87712

**Pull request closed without merging**
